src/openms/source/FORMAT/QcMLFile.cpp

```diff
@@ -1202,6 +1272,150 @@ namespace OpenMS
+        std::map<String, std::vector<QualityParameter> >::const_iterator qpsit = runQualityQPs_.find(*it);
+        if (qpsit != runQualityQPs_.end())
+        { 
+          for (std::vector<QcFile::QualityParameter>::const_iterator qit = qpsit->second.begin(); qit != qpsit->second.end(); ++qit)
+          { 
+            qualityMetrics += qit->toJSONString();
+            if (qit->name == "instrument model") 
+            {
+              instrumentModel = qit->value;
+            } else if (qit->name == "completion time")
+            {
+              completionTime = qit-> value;
+            }
+          }
+        }
+        // adding QualityAttachments to qualityMetrics
+        std::map<String, std::vector<Attachment> >::const_iterator attit = runQualityAts_.find(*it);
+        if (attit != runQualityAts_.end())
+        {
+          for (std::vector<QcFile::Attachment>::const_iterator ait = attit->second.begin(); ait != attit->second.end(); ++ait)
+          {
+            qualityMetrics += ait->toJSONString(); //TODO check integrity of reference to qp!
+          }
+        }
+      }
+    // removing trailing ",\n" from last qualityMetric
+    qualityMetrics.pop_back();
+    qualityMetrics.pop_back();
+
+    ofstream os(outputfilename.c_str());
+    if (!os)
+    {
+      throw Exception::UnableToCreateFile(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, outputfilename);
+    }
+
+    os.precision(writtenDigits<double>(0.0));
+
+    
+    os << String("{ \"mzQC\": {\n")
+        << String(4, ' ') << "\"creationDate\": \"" << put_time(ltm, "%Y-%m-%dT%H:%M:%S") <<"\",\n"
+        << String(4, ' ') << "\"version\": \"1.0.0\",\n"
+        // runQualities
+        << String(4, ' ') << "\"runQualities\": [\n"
+        << String(6, ' ') << "{\n"
+        << String(8, ' ') << "\"metadata\": {\n"
+        // runQualities metadata input files and file properties
+        << String(10, ' ') << "\"inputFiles\": [\n"
+        << String(12, ' ') << "{\n"
+        << String(14, ' ') << "\"location\": \"" << outputFilePath << "\",\n"
+        << String(14, ' ') << "\"name\": \"" << inputFileName << "\",\n"
+        << String(14, ' ') << "\"fileFormat\": {\n"
+        << String(16, ' ') << "\"accession\": \"MS:1000584\",\n" 
+        << String(16, ' ') << "\"name\": \"mzML format\"\n"
+        << String(14, ' ') << "},\n"
+        << String(14, ' ') << "\"fileProperties\": [\n"
+        << String(16, ' ') << "{\n"
+        << String(18, ' ') << "\"accession\": \"MS:1000747\",\n" // exclude in qualityMetrics?
+        << String(18, ' ') << "\"name\": \"completion time\",\n"
+        << String(18, ' ') << "\"value\": \"" + completionTime + "\"\n"
+        << String(16, ' ') << "},\n"
+        << String(16, ' ') << "{\n"
+        << String(18, ' ') << "\"accession\": \"MS:1000569\",\n"
+        << String(18, ' ') << "\"name\": \"SHA-1\",\n"
+        << String(18, ' ') << "\"value\": \"" << "\"\n" // how to calculate SHA sum?
```

Comment:
from MSExperiment you can get the SourceFile https://github.com/OpenMS/OpenMS/blob/develop/src/tests/class_tests/openms/source/SourceFile_test.cpp#L150